src/topology/metric_space/emetric_space.lean

```diff
@@ -740,6 +740,33 @@ begin
     exact not_lt.1 (λ hlt, Hgt I.1 hlt I.2 hI.some_spec) }
 end
 
+/-- For a set `s` in a pseudo emetric space, if for every `ε > 0` there exists a countable
+`ε`-dense set in `s`, then there exists a countable subset `t ⊆ s` that is dense in `s`. -/
+lemma subset_countable_closure_of_almost_dense_set (s : set α)
+  (hs : ∀ ε > 0, ∃ t ⊆ s, countable t ∧ s ⊆ ⋃ x ∈ t, closed_ball x ε) :
```

Comment:
The assumption can be weakened: you don't need to assume that `t` is included in `s` (by taking a point in `s \cap closed_ball x epsilon` if the ball is not empty, and then replacing epsilon by 2epsilon).